### ISSUE TITLE
machine credential type disapeared in 17.0.1 version

### ISSUE THREAD

**Issue by abdel-ops** [ansible/awx#9610]
##### ISSUE TYPE
 - Bug Report

##### ENVIRONMENT
* AWX version: X.Y.Z
* AWX install method: openshift, minishift, docker on linux, docker for mac, boot2docker
* Ansible version:  X.Y.Z
* Operating System:
* Web Browser:

**Comment by abdel-ops** [comment]
Hello
After i have installed awx in version 17.0.1 , i noticed that i the credential type 'Machine' don't appear anymore in the list of credential type list displayed when we want to create a new credential.
![image](https://user-images.githubusercontent.com/60286045/111523652-d1689d80-875b-11eb-9a43-bfc823d5c128.png)
is it been deprecated ?
thanks for your help

**Comment by unlikelyzero** [comment]
@abdel-ops which version did you upgrade from? Maybe it was an incomplete migration? I see it here:
![Screen Shot 2021-03-18 at 4 23 48 PM](https://user-images.githubusercontent.com/12446869/111692891-58725000-8806-11eb-8f89-ccbf6ea39bc3.png)

**Comment by dannyvanderaa** [comment]
Same issue here, seems to be for all Tower managed credential types. For example i am also missing source control

**Comment by mabashian** [comment]
@abdel-ops @dannyvanderaa do you all have some custom credential types?  If so, how many?

**Comment by dannyvanderaa** [comment]
@mabashian , yes I have Custom Credential Types, 16 :)

**Comment by abdel-ops** [comment]
For me it noticed that i loose machine and source control credential type after i performed an import  with awx cli command (awx --import) from awx 6.0.0 to 17.0.1

**Comment by ryanpetrello** [comment]
@abdel-ops did you *upgrade* directly from 6.0.0 to 17.0.1?

**Comment by ryanpetrello** [comment]
I'm not sure how `awx --import` could get your install into this state, because the `Machine` credential shouldn't be deletable from the API.  I suspect you followed an upgrade path across multiple versions that we just haven't tested, and are encountering some sort of bug.

**Comment by dannyvanderaa** [comment]
I upgraded from 15.0.1 to 17.1.0.
Can i do something to support you, to get this solved?

**Comment by donald-talbert989** [comment]
We are on 17.1.0 and upgraded from 15.0.1 too. I believe this issue may not be related to the upgrade path though. I believe its a pagination issue related to the number of credential types present on the instance.  It seems the new UI is not parsing all of the pages of the `/api/v2/credential_types` endpoint (See screenshot). 

> Here I am at the /#/credentials/add page. The 2nd page of the request is never loaded in.
![image](https://user-images.githubusercontent.com/81595827/112912307-3eb5f000-90c5-11eb-8e14-bd3a0ed290f6.png)


Right now three of our instances have this issue, but one of our test instances doesn't have the issue. The difference between the test and other instances is that the test instance only has a small number credential types present so all fit on the first page.

**Comment by quasd** [comment]
Quick dirty fix, change PAGE_SIZE 25 -> 50
/var/lib/awx/venv/awx/lib/python3.6/site-packages/awx/settings/defaults.py
```
    'PAGE_SIZE': 50,
```

**Comment by dannyvanderaa** [comment]
Unfortenately this didn't solve this issue.
Still can't choose credential types like machine and source control, when adding credentials

**Comment by quasd** [comment]
> Unfortenately this didn't solve this issue.
> Still can't choose credential types like machine and source control, when adding credentials

Did you restart the service/docker? And ofc the 50 limit might have to be higher if you have more credential types. If on docker this needs to be changed inside awx_web

**Comment by dannyvanderaa** [comment]
Restart of the docker did the trick (forgot that).
Hopefully this gets a permanent fix in a new release.

**Comment by donald-talbert989** [comment]
> Quick dirty fix, change PAGE_SIZE 25 -> 50
> /var/lib/awx/venv/awx/lib/python3.6/site-packages/awx/settings/defaults.py
> 
> ```
>     'PAGE_SIZE': 50,
> ```

Thanks for the temporary workaround @quasd . Changing the value to 50 and restarting the docker container worked.

**Comment by ryanpetrello** [comment]
@jakemcdermott do you have any ideas here?  The suggested fix above makes me wonder if the UI has a bug that doesn't properly page through all credential types.

**Comment by unlikelyzero** [comment]
@ryanpetrello that's the issue OR these users are not system admin users and we're pulling the wrong RBAC endpoint.

In our tests, we only have 5-10 credentials at a given time OR we mock out 200+

**Comment by ghost** [comment]
We're using a system administrative user and not seeing some of the AWX Managed credential types in the dropdowns.

Also, if we try and edit a Managed credential type like 'Machine' or 'Source Control' that's not in the first 25 returned credential types, the page goes white and throws a javascript error. 

We have 13 custom credential types that are all visible. In all Credential Type dropdowns, we're seeing only the first 25 (our 13 custom and the last 12 rows of the built-in types.

**Comment by phospi** [comment]
> Quick dirty fix, change PAGE_SIZE 25 -> 50
> /var/lib/awx/venv/awx/lib/python3.6/site-packages/awx/settings/defaults.py
> 
> ```
>     'PAGE_SIZE': 50,
> ```

@quasd You Sir are my hero and savior.

Situation: 
We did an upgrade vom 15.0.1 to 17.1.0. No versions have been left out and we noticed to things:
1. Editing machine credentials was not possible. The website turned white. No errors in `docker logs`
2. When creating a new credential no built-in was shown in dropdown. 

Solution:
We turned the page size up like described and everything worked out fine. 
1. Machine credentials are editable again.
2. When creating new credentials all builtin-in credentials are now shown.
